assembly/src/assembler/instruction/mod.rs

```diff
@@ -329,3 +329,17 @@ fn push_felt(span: &mut SpanBuilder, value: Felt) {
         span.push_op(Push(value));
     }
 }
+
+/// Returns an error if the specified value is smaller than or equal to min or greater than or
+/// equal to max. Otherwise, returns Ok(()).
+fn validate_param<I: Ord + Into<u64>>(value: I, min: I, max: I) -> Result<(), AssemblerError> {
```

Comment:
non-blocking nit: we could in the future check this with [Range](https://doc.rust-lang.org/std/ops/struct.Range.html) boundaries so the logic for inclusive/exclusive ranges will be unmistakable

Reply:
Great idea! Let's create an issue for this.